### PR TITLE
Expand APP_CONTEXT_BASE_DIRECTORY to workaround Mono issues.

### PR DESCRIPTION
**Pull request by pranavkm**
See https://github.com/aspnet/Configuration/pull/354#issuecomment-165080268

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl

**Comment by davidfowl** [comment]
One thing that you should test, are app_base_path and appdomain base the same?

**Comment by pranavkm** [comment]
This doesn't work. We need to target netcore50 to allow targeting Configuration. However `System.Runtime.InteropServices.RuntimeInformation` doesn't have an RTM package. I'm just going to use the existing code in configuration for now.

**Comment by anurse** [comment]
Seems fine to me but I don't have the full context. What the heck, :shipit: anyway :)

File: src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs [hunk 1]
```diff
@@ -55,6 +34,18 @@ public FrameworkName RuntimeFramework
             }
         }
 
+        private static string GetApplicationBasePath()
+        {
+            var basePath =
+#if NET451
+                (string)AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") ??
```
Comment (davidfowl) [inline src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs]: Path.GetFullPath?

File: src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs [hunk 2]
```diff
@@ -55,6 +34,18 @@ public FrameworkName RuntimeFramework
...
+                AppDomain.CurrentDomain.BaseDirectory;
+#else
+                AppContext.BaseDirectory;
+#endif
+            return Path.GetFullPath(basePath);
```
Comment (davidfowl) [inline src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs]: Does this do what we think it does? Can you print out what this is for a vanilla mono app compared to AppDomain.CurrentDomain.BaseDirectory ?

Comment (pranavkm) [inline src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs]: ~/src/ConsoleApp2/src/ConsoleApp2$ dnx run
APP_CONTEXT_BASE_DIRECTORY: .
Expanded APP_CONTEXT_BASE_DIRECTORY: /home/prkrishn/src/ConsoleApp2/src/ConsoleApp2
AppDomain.BasePath: /home/prkrishn/.dnx/runtimes/dnx-mono.1.0.0-rc2-16249/bin/
Current Directory:/home/prkrishn/src/ConsoleApp2/src/ConsoleApp2

~/src/ConsoleApp2/src/ConsoleApp2$ mono Test.exe
APP_CONTEXT_BASE_DIRECTORY: 
AppDomain.CurrentDomain.BaseDirectory: /home/prkrishn/src/ConsoleApp2/src/ConsoleApp2/

using Mono JIT compiler version 4.2.1

Comment (davidfowl) [inline src/Microsoft.Extensions.PlatformAbstractions/DefaultApplicationEnvironment.cs]: Great!

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**